addons/hw_drivers/iot_handlers/drivers/DisplayDriver_L.py

```diff
@@ -86,52 +94,68 @@ def load_url(self):
             urllib3.disable_warnings()
             http = urllib3.PoolManager(cert_reqs='CERT_NONE')
             try:
-                response = http.request('GET', "%s/iot/box/%s/display_url" % (helpers.get_odoo_server_url(), helpers.get_mac_address()))
+                response = http.request(
+                    'GET',
+                    "%s/iot/box/%s/display_url" % (helpers.get_odoo_server_url(), helpers.get_mac_address())
+                )
                 if response.status == 200:
                     data = json.loads(response.data.decode('utf8'))
                     url = data[self.device_identifier]
             except json.decoder.JSONDecodeError:
                 url = response.data.decode('utf8')
             except Exception:
                 pass
-        return self.update_url(url)
 
-    def call_xdotools(self, keystroke):
-        os.environ['DISPLAY'] = ":0." + self._x_screen
-        os.environ['XAUTHORITY'] = "/run/lightdm/pi/xauthority"
-        try:
-            subprocess.call(['xdotool', 'search', '--sync', '--onlyvisible', '--screen', self._x_screen, '--class', 'Firefox', 'key', keystroke])
-            return "xdotool succeeded in stroking " + keystroke
-        except:
-            return "xdotool threw an error, maybe it is not installed on the IoTBox"
+        return url
 
     def _action_update_url(self, data):
         if self.device_identifier != 'distant_display':
             self.update_url(data.get('url'))
 
     def _action_display_refresh(self, data):
         if self.device_identifier != 'distant_display':
-            self.call_xdotools('F5')
+            self.browser.refresh()
+
+    def _set_environ(self):
+        os.environ['DISPLAY'] = ":0." + self._x_screen
+        os.environ['XAUTHORITY'] = '/run/lightdm/pi/xauthority'
+
+        return os.environ.copy()
+
+    def set_orientation(self, orientation=Orientation.NORMAL):
+        subprocess.run(['xrandr', '-o', orientation.value], check=True)
+        subprocess.run([file_path('hw_drivers/tools/sync_touchscreen.sh'), str(int(self._x_screen) + 1)], check=False)
+        helpers.save_browser_state(orientation=orientation)
+
 
 class DisplayController(http.Controller):
     @http.route('/hw_proxy/customer_facing_display', type='json', auth='none', cors='*')
-    def customer_facing_display(self):
-        data = http.request.get_json_data()
-        if data['action'] == 'open':
+    def customer_facing_display(self, action, pos_id=None, access_token=None, data=None):
+        display = self.ensure_display()
+        if action in ['open', 'open_kiosk']:
             origin = helpers.get_odoo_server_url() or http.request.httprequest.referrer.split('://')[-1].split('/')[0]
```

Comment:
so assuming you can’t trust the referer, it would be a way to inject malicious url in the configuration file for the next person?
update_url -> open_browser -> save_browser_state -> write_file('browser-url.conf', url)

Reply:
@mart-e now that I removed the possibility to get the origin from the referrer, the url can only be the db's one.